### PR TITLE
fix(loop): resolve repo root/ledger worktree-relative, not from cwd (#1052, #1019, #1050)

### LINKED ISSUE #1019: detect-copilot-loop-state resolves repo root from process.cwd() — maxCopilotRounds config silently falls back to default 5 when cwd lacks .devloops
## Problem

`detect-copilot-loop-state.mjs` resolves the repo root as `path.resolve(process.cwd())` when loading dev-loop config, so `refinement.maxCopilotRounds` is read from wherever the detector happens to be invoked — not from the actual repository root. When the detector runs from a cwd that doesn't contain `.devloops` (e.g. the main checkout when `.devloops` lives only in a worktree, or any non-repo-root cwd), it falls back to the built-in default `maxCopilotRounds: 5`. This makes the round-cap enforcement in `interpretLoopState` use the wrong cap, so a PR that has actually exhausted its configured Copilot rounds is misreported as `autoRerequestEligible: true` under `state: ready_to_rerequest_review` instead of the correct `round_cap_clean_fallback`.

## Reproduction (live, on PR #123 mfittko/repo-wiki)

Repo config (in `tmp/worktrees/dev-loops/pr-123/.devloops`) sets `refinement.maxCopilotRounds: 2`. The PR has 2 completed Copilot rounds, 0 unresolved threads, green CI, head advanced past the last Copilot review.

Run the detector from the main checkout (no `.devloops` there):

```sh
node <pkg>/scripts/loop/detect-copilot-loop-state.mjs --repo mfittko/repo-wiki --pr 123
```

Actual output:
```json
{
  "state": "ready_to_rerequest_review",
  "autoRerequestEligible": true,
  "snapshot": {
    "copilotReviewRoundCount": 2,
    "copilotReviewRequestStatus": "none",
    "unresolvedThreadCount": 0,
    "ciStatus": "success",
    "copilotReviewOnCurrentHead": false
  }
}
```

Expected (with the repo's `maxCopilotRounds: 2` honored): `state: round_cap_clean_fallback`, `autoRerequestEligible: false`, `roundCapCleanEligible: true`.

Running the same detector from inside the worktree that contains `.devloops` (so `process.cwd()` has the config) produces the correct `round_cap_clean_fallback` — confirming the bug is the cwd-based repo-root resolution, not the round-cap logic itself.

## Root cause (source-verified)

`scripts/loop/detect-copilot-loop-state.mjs:448`:

```js
const config = await loadDevLoopConfig({ repoRoot: path.resolve(process.cwd()) });
const refinementConfig = config.errors.length > 0
  ? resolveRefinement({ version: 1 })
  : resolveRefinement(config.config);
interpretation = interpretLoopState(interpretationInput, refinementConfig);
```

`path.resolve(process.cwd())` is the cwd, not the repo root. When `.devloops` is absent from the cwd (which is the common case — the main checkout has no `.devloops`, only the worktree does, and the detector is often invoked from the main checkout or a harness context whose cwd is not the worktree root), `loadDevLoopConfig` returns `config.errors` non-empty → the detector falls back to `resolveRefinement({ version: 1 })` (pure defaults: `maxCopilotRounds: 5`).

The round-cap block in `@dev-loops/core/src/loop/copilot-loop-state.mjs` (around line 375) correctly consults `refinementConfig?.maxCopilotRounds`, and `isAutoRerequestEligible` correctly returns `false` for `ROUND_CAP_CLEAN_FALLBACK` — so the *state machine* is correct. The bug is purely that the detector feeds it the wrong cap because it resolves the config from the wrong root.

Compare the startup resolver (`resolve-dev-loop-startup.mjs`), which resolves the repo root via the adapter (`adapter.getRepoRoot()`) rather than `process.cwd()` — the detector is inconsistent with that resolution path.

Secondary contributor: there is no mechanism that ensures `.devloops` set in a worktree is visible to detector invocations from outside that worktree. The worktree-local `.devloops` is effectively invisible to any cwd-based config read, which is how this repo got into the `maxCopilotRounds: 2`-only-in-worktree state in the first place.

## Impact

- A PR that has legitimately exhausted its configured Copilot rounds is reported as `autoRerequestEligible: true` and `state: ready_to_rerequest_review`, inviting an illegal over-cap Copilot re-request.
- The round-cap safety property (stop re-requesting after the configured cap) is silently bypassed whenever the detector's cwd differs from the `.devloops` location.
- Operator/agent scripts that call the detector from a non-worktree cwd get a misleadingly optimistic loop state.

## Proposed fix

Resolve the repo root the same way the startup resolver does — via the harness adapter (`adapter.getRepoRoot()`) or a git-toplevel resolution — instead of `path.resolve(process.cwd())`. Concretely in `detect-copilot-loop-state.mjs`:

```js
// Before
const config = await loadDevLoopConfig({ repoRoot: path.resolve(process.cwd()) });

// After — resolve actual repo root (git toplevel fallback when no adapter)
const repoRoot = await resolveRepoRoot(); // git rev-parse --show-toplevel, or adapter.getRepoRoot()
const config = await loadDevLoopConfig({ repoRoot });
```

Additionally, consider: when `loadDevLoopConfig` returns config errors (no `.devloops` found at the resolved repo root), surface that as a distinct signal rather than silently falling back to defaults, so a missing-config condition is visible rather than masquerading as `maxCopilotRounds: 5`.

## Environment

- dev-loops `0.5.0`, @dev-loops/core `0.5.0`
- Repo `mfittko/repo-wiki`, PR #123, `.devloops` present only in the worktree (`tmp/worktrees/dev-loops/pr-123/.devloops`), absent from the main checkout.

## Acceptance criteria

- [x] `detect-copilot-loop-state.mjs --repo <owner/name> --pr <n>` reports the correct `maxCopilotRounds`-derived state regardless of the invoking cwd (i.e. running it from the main checkout produces the same result as running it from the worktree that holds `.devloops`).
- [x] A repo with `refinement.maxCopilotRounds: 2`, 2 completed Copilot rounds, 0 unresolved threads, green CI, and an advanced head reports `state: round_cap_clean_fallback` + `autoRerequestEligible: false` when the detector is invoked from any cwd.
- [x] Regression test: invoke the detector from a cwd without `.devloops` and assert it still resolves the repo-root config (not the built-in default).

### LINKED ISSUE #1050: Pre-merge fan-out ledger check binds to session cwd, not the PR worktree — false merge block across worktrees
## Problem

The pre-merge fan-out evidence check (`gates.requireFanoutEvidence`) verifies the findings-log **ledger** exists on disk, but resolves that ledger path against `process.cwd()` — while the gate that *wrote* the ledger also wrote it cwd-relative. When the gate work and the `gh pr merge` run from **different checkouts/worktrees**, the two `tmp/` paths diverge and the merge is blocked even though the gates genuinely ran clean.

Hit live merging PR #1044: draft_gate (15 angles) + pre_approval_gate (14 angles) both ran clean via `fanout_fanin` in a dedicated PR worktree (`tmp/worktrees/dev-loops/pr-1044/tmp/gate-findings/...`), but `gh pr merge` was blocked twice by the bash gate hook. Root cause: the hook resolves `repoRoot` from the **session cwd** (a *different* worktree) and runs `detect-checkpoint-evidence.mjs` from there, where the ledger `tmp/` file did not exist. Workaround was to copy the two ledger JSONs into the session-cwd worktree's `tmp/`.

## Root cause (traced)

- `scripts/github/detect-checkpoint-evidence.mjs` `buildFanoutEnforcement()` builds `ledgerPath = buildLogPath({..., tmpRoot: "tmp"})` then checks `path.resolve(cwd, ledgerPath)` — `cwd` is the process working dir (the hook passes the session `repoRoot`).
- `scripts/github/write-gate-findings-log.mjs` `writeGateFindingsLog()` writes to `process.cwd()`-relative `tmp/gate-findings/...`.
- `.claude/hooks/pre-tool-use-bash-gate.mjs` runs the evidence script with `cwd: repoRoot`, where `repoRoot = git rev-parse --show-toplevel` of the **session cwd** — not the PR worktree, not the `cd` target inside the merge command.

So ledger-write and ledger-check are both cwd-relative but bind to *different* checkouts in the normal per-PR-worktree flow. `tmp/` is gitignored scratch (correctly), so the two silently diverge.

Same class of bug as #1019 (detect-copilot-loop-state resolving repo root from `process.cwd()`).

## Impact

Any dev-loop run that does gate work in a per-PR worktree (the documented/normal flow) but issues `gh pr merge` from a session sitting in a different worktree gets a false merge block. The gates are clean; only the ledger-location binding is wrong. Fail-closed is correct in spirit, but here it fails closed on a false negative.

## Acceptance criteria

- [x] The pre-merge fan-out ledger existence check locates the ledger written by the clean gate run regardless of which checkout/worktree the merge command runs from (for the same repo + PR + head SHA).
- [x] A gate run in a per-PR worktree followed by `gh pr merge` from a different worktree of the same repo passes the evidence check without manually copying `tmp/` files.
- [x] Fail-closed behavior is preserved when the ledger genuinely does not exist for the reviewed head.

## Definition of done

- Ledger location is resolved deterministically per (repo, PR, head SHA) rather than per arbitrary session cwd — e.g. resolve the ledger against the PR's own worktree/checkout, or store it at a repo-shared location keyed by repo+PR, or have the hook resolve the correct PR checkout before running the evidence script.
- One runnable check covering the cross-worktree case.

## Non-goals

- Reworking the fan-out evidence contract itself (`requireFanoutEvidence` semantics stay).
- Making `tmp/` tracked in git.
- Broader cwd-resolution audit beyond this ledger/hook path (though #1019 is a related sibling worth bundling if convenient).

## Notes

Fix is a real design choice (shared ledger location vs. PR-worktree resolution vs. hook-side checkout resolution), not a one-liner — filing rather than patching inline.

### LINKED ISSUE #1052: fix(loop): resolve repo root worktree-relative, not from process.cwd() (umbrella: #1019, #1050)
## Problem (umbrella)

Multiple loop scripts resolve the repository root / read state from `process.cwd()`, which is wrong under the normal per-PR-worktree flow: the session cwd and the PR worktree are different checkouts, so cwd-bound reads/writes silently diverge. Two confirmed symptoms:

- **#1050** — the pre-merge fan-out ledger check binds to session cwd, not the PR worktree, producing a false merge block on clean gates (hit PR #1044 twice; ledgers had to be hand-copied to unblock).
- **#1019** — `detect-copilot-loop-state` resolves repo root from `process.cwd()`, so `maxCopilotRounds` silently falls back to the default 5 when cwd lacks `.devloops`.

Both share one root cause: cwd-relative resolution where PR/worktree-relative is required.

## Proposed approach

1. Audit all `process.cwd()`-based repo-root / `.devloops` / `tmp/`-ledger resolution across `scripts/loop/**`, `scripts/github/**`, and `.claude/hooks/**`.
2. Establish one shared resolver that derives the repo root from the PR/worktree under operation (explicit `--repo-root`/`--worktree`, or git-toplevel of the file being acted on), not the ambient cwd.
3. Fix the two known sites (#1019, #1050) and any others the audit surfaces.
4. Ensure the pre-merge bash-gate hook and the ledger writer resolve to the SAME checkout so evidence written by the gates is always visible to the merge check.

## Acceptance criteria

- Fan-out ledgers written during gates in a PR worktree are found by the pre-merge check regardless of the session's cwd (closes #1050).
- `maxCopilotRounds` (and other `.devloops` config) resolves from the PR worktree, not cwd (closes #1019).
- An audit list of remaining cwd-based resolutions with each either fixed or explicitly justified.
- Regression tests exercising the cross-worktree case (write in worktree A, check from cwd B).

## Non-goals

- Changing gate semantics or the ledger format; this is purely resolution-location correctness.

Folds in #1019 and #1050.

### PR DESCRIPTION
**Pull request by mfittko**
Closes #1052
Closes #1019
Closes #1050

## Summary

Multiple loop scripts resolved the repo root / `.devloops` / `tmp/` ledger location from `process.cwd()`. Under the per-PR-worktree flow the session cwd (and its git-toplevel) is a *different* checkout than the PR worktree, so cwd-bound reads silently diverged from cwd-bound writes. This umbrella establishes one shared worktree-relative resolver and fixes the confirmed symptoms. The resolution-location correctness is the theme; the one intentional gate-resolution refinement (umbrella multi-close) is called out explicitly below.

## Scope

- New shared resolver `scripts/loop/_repo-root-resolver.mjs`: `resolveRepoRoot(cwd)` (git-toplevel of the checkout under operation, not ambient cwd) and `resolveLedgerCheckouts(cwd)` (main checkout + every worktree, so a ledger written in any checkout is visible from any other).
- **#1050**: the pre-merge fan-out ledger check now looks for the ledger across all checkouts of the repo, so a ledger written in a PR worktree is visible to the merge check even when the bash-gate hook runs from a different session checkout. No more false "missing pre-merge gate evidence" block on clean gates.
- **#1019**: repo root is resolved worktree-relative before loading config, so `.devloops` (and `maxCopilotRounds`) resolve from the PR worktree instead of silently falling back to the default 5. Fixed in `detect-copilot-loop-state` and its clone in `copilot-pr-handoff`.

## File-by-file

- `scripts/loop/_repo-root-resolver.mjs` — new shared resolver (reuses `parseAllWorktreePaths` / `parseMainWorktreePath` from `@dev-loops/core/loop/worktree-guard`).
- `scripts/github/detect-checkpoint-evidence.mjs` — `buildFanoutEnforcement` checks the ledger across `resolveLedgerCheckouts(cwd)` instead of the single `path.resolve(cwd, ledgerPath)`.
- `scripts/loop/detect-copilot-loop-state.mjs` — load config from `resolveRepoRoot(process.cwd())` instead of raw `process.cwd()`.
- `scripts/loop/copilot-pr-handoff.mjs` — route config load and runner-ownership `cwd` through `resolveRepoRoot(process.cwd())` (same #1019 root-cause class).
- `scripts/loop/detect-pr-gate-coordination-state.mjs` — the two `runtime.repoRoot ?? process.cwd()` fallbacks now fall back to `resolveRepoRoot(process.cwd())`; the umbrella multi-close refinement resolution (see below); the mixed partial-fetch-failure branch now reports against the first successfully-fetched issue rather than `evaluated[0]` (which may be a failed fetch).
- Tests: cross-worktree ledger regression + resolver unit tests + worktree-relative `maxCopilotRounds`/`.devloops` resolution + umbrella refinement cases (present / all-missing / mixed partial-failure).

## Audit — process.cwd()-based repo-root/ledger resolution

Per #1052 AC #1, every `process.cwd()`-based repo-root/ledger resolution in `scripts/loop/**`, `scripts/github/**`, `.claude/hooks/**` is either fixed or justified:

**FIXED**

- `scripts/loop/detect-copilot-loop-state.mjs` — config `repoRoot` → `resolveRepoRoot`.
- `scripts/loop/copilot-pr-handoff.mjs` — config load + runner-ownership `cwd` → `resolveRepoRoot`.
- `scripts/loop/detect-pr-gate-coordination-state.mjs:464,518` — `runtime.repoRoot` fallbacks → `resolveRepoRoot(process.cwd())` (explicit `runtime.repoRoot` path was already correct; only the cwd fallback was the hazard).
- `scripts/github/detect-checkpoint-evidence.mjs` `buildFanoutEnforcement` — ledger lookup → `resolveLedgerCheckouts` across all checkouts.
- `scripts/github/detect-checkpoint-evidence.mjs:395` — config load `repoRoot` → `resolveRepoRoot(cwd)` (same #1019 divergence class as the ledger read in the same function; a raw cwd there would fail open and silently disable the fan-out evidence check).
- `.claude/hooks/pre-tool-use-bash-gate.mjs` + `write-gate-findings-log.mjs` — the cross-worktree ledger divergence, fixed on the read side via `resolveLedgerCheckouts` so the writer's worktree ledger is always visible to the reader.

**JUSTIFIED / OUT OF SCOPE**

- `scripts/loop/detect-change-scope.mjs:100` — runs in the local-implementation worktree by contract; cwd IS the worktree root, so no divergence.
- `scripts/loop/inspect-run-viewer/server.mjs` cwd reads (resolver target + config load) — operator-launched local inspect-run viewer; cwd = the launch dir the operator ran it from, not the per-PR-worktree hook/gate flow, so there is no cwd/worktree divergence to resolve here.
- `cwd = process.cwd()` DEFAULT PARAMETERS in `_pr-runner-coordination.mjs`, `_stale-runner-detection.mjs`, `steer-loop.mjs`, `conductor*.mjs`, `info.mjs`, `inspect-run-viewer/server.mjs`, etc. — injectable defaults whose callers pass an explicit `cwd`/`repoRoot`; not the divergence bug.
- The shared resolver is now available for future consolidation of the inline `git rev-parse --show-toplevel` sites (`run-refinement-audit`, `resolve-dev-loop-startup`, `ensure-worktree`, the bash-gate hook). That broad consolidation is explicitly a non-goal of this PR.

## Intentional gate-resolution change

The draft-gate refinement guard now resolves umbrella PRs that close multiple issues. Previously it required EXACTLY one closing reference, which false-blocked umbrella PRs. A PR's scope is refinement-`present` if ANY closing-referenced issue carries the refinement artifact (ACs/DoD) — the any-of contract; `refinedIssues` records which linked issues satisfied it. This is linked-issue *resolution* correctness (the theme of this PR), not a change to what the gate requires of a refined issue: the ledger format and pre-merge failure strings are unchanged.

## Acceptance

- [x] One shared resolver derives repo root from the checkout under operation, not ambient cwd
- [x] #1050: pre-merge ledger written in worktree A is found when the check runs from checkout B
- [x] #1019: `maxCopilotRounds` resolves from the worktree `.devloops`, not the cwd fallback default (both `detect-copilot-loop-state` and `copilot-pr-handoff`)
- [x] Regression test covers the cross-worktree write-A/check-B case
- [x] Audit list of remaining `process.cwd()`-based resolutions, each fixed or justified (above)
- [x] Umbrella any-of refinement resolution covered by tests (present / all-missing / mixed partial-failure)

## Definition of done

- [x] Shared resolver module added and reused by the fixed sites
- [x] Confirmed #1019/#1050 sites fixed plus the audited clones
- [x] Regression + umbrella refinement tests green
- [x] `npm run verify` green
- [x] Draft gate + pre-approval gate posted; Copilot review addressed

## Non-goals

- Wiring every `process.cwd()` caller across the codebase through the resolver (only the confirmed symptoms + the shared resolver are in scope; the rest are audited as justified/out-of-scope above).
- Broad consolidation of the inline `git rev-parse --show-toplevel` sites onto the shared resolver.
- PR→branch→worktree network lookup: the resolver enumerates local checkouts, which covers the divergence deterministically without a network call.

## Validation

- `npm run verify`
- Cross-worktree regression test (write ledger in worktree A, run pre-merge check from checkout B → found).
- Umbrella refinement cases: any-of present, all-missing, and mixed partial-fetch-failure.

### CODE REVIEW

**Comment by mfittko** [comment]
### Gate review: `draft_gate`

**Reviewed head SHA:** `6777346b7ee953061e7573841404a9d985da6d3a`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 3 angles reviewed; 7 findings (see per-angle breakdown below).

- `correctness` → clean
  - [worth-fixing-now] Umbrella multi-close was an undocumented gate-semantics change contradicting the stated non-goal; reconciled in PR body as an intentional linked-issue-resolution change. — _accepted-for-fix_
  - [worth-fixing-now] any-of refinement contract could pass an unrefined closing sibling; documented as intentional umbrella semantics with refinedIssues recording which linked issues satisfied it. — _accepted-for-fix_
  - [defer] resolveLedgerCheckouts de-dup relies on git canonical paths; acceptable, low risk. — _deferred_
- `scope` → clean
  - [must-fix] #1019 clone in copilot-pr-handoff.mjs (config + runner-ownership cwd from raw process.cwd()); routed through resolveRepoRoot. — _accepted-for-fix_
  - [must-fix] Missing audit-list AC; added process.cwd() audit section (FIXED / JUSTIFIED-OUT-OF-SCOPE) to PR body. — _accepted-for-fix_
  - [worth-fixing-now] detect-pr-gate-coordination-state.mjs cwd fallbacks (lines 464,518) hardened via resolveRepoRoot. — _accepted-for-fix_
- `coverage` → clean
  - [worth-fixing-now] Mixed partial-fetch-failure branch untested; added test which surfaced and root-caused a null-deref crash (report against first fetched issue). — _accepted-for-fix_

**Next action:** mark ready for review; proceed to Copilot review then pre_approval_gate

File: scripts/loop/_repo-root-resolver.mjs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const roots = [];
+  const add = (p) => { if (typeof p === "string" && p.length > 0 && !roots.includes(p)) roots.push(p); };
+  add(resolveRepoRoot(cwd, { gitCommand }));
+  try {
+    const listing = execFileSync(gitCommand, ["worktree", "list"], {
+      cwd, encoding: "utf8", stdio: ["ignore", "pipe", "ignore"],
+    });
+    add(parseMainWorktreePath(listing));
+    for (const p of parseAllWorktreePaths(listing)) add(p);
+  } catch { /* git unavailable: cwd-toplevel is all we have */ }
```
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/loop/_repo-root-resolver.mjs]: Fixed in 62886b6d. Reworded the resolveRepoRoot catch comment to "git failed or cwd is not inside a git repo (or git is unavailable) -> fall back to cwd", and applied the same clarification to the resolveLedgerCheckouts catch comment. The over-specific "git unavailable" claim is gone.

File: scripts/loop/detect-pr-gate-coordination-state.mjs [hunk 2]
```diff
@@ -262,58 +276,117 @@ async function fetchIssueBody({ repo, issue }, { env = process.env, ghCommand =
...
-  if (linkedIssue === null) {
+export async function loadRefinementArtifact({ repo, prData, prDraft, prClosed, prMerged }, { env = process.env, ghCommand = "gh" } = {}) {
+  const linkedIssues = resolveLinkedIssuesFromPr(prData);
+  if (linkedIssues.length === 0) {
     if (prDraft) {
       return {
         status: "missing",
         linkedIssue: null,
+        linkedIssues: [],
         reason: "Draft PR has no deterministically resolvable linked issue (no closingIssuesReferences, no unique Closes/Fixes/Resolves pattern in body); draft gate cannot verify a refinement artifact.",
```
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/loop/detect-pr-gate-coordination-state.mjs]: Fixed in 62886b6d. Dropped the stale "unique" wording: the draft-branch reason string now reads "no closingIssuesReferences and no Closes/Fixes/Resolves #n reference in body". resolveLinkedIssuesFromPr accepts multiple closing refs, so uniqueness no longer applies. No change to finding value or control flow.

File: scripts/loop/detect-pr-gate-coordination-state.mjs [hunk 3]
```diff
@@ -262,58 +276,117 @@ async function fetchIssueBody({ repo, issue }, { env = process.env, ghCommand =
...
         finding: "missing_refinement_artifact",
       };
     }
     return {
       status: "unknown",
       linkedIssue: null,
+      linkedIssues: [],
       reason: "No deterministically resolvable linked issue (no closingIssuesReferences, no unique Closes/Fixes/Resolves pattern in body).",
```
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/loop/detect-pr-gate-coordination-state.mjs]: Fixed in 62886b6d. Same de-staling applied to the non-draft/unknown-branch reason string: now "no closingIssuesReferences and no Closes/Fixes/Resolves #n reference in body". Updated the matching test expectation. No change to finding value or control flow.

File: test/loop/repo-root-resolver.test.mjs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+
+test("resolveLedgerCheckouts falls back to cwd when git is unavailable", async () => {
+  const dir = await realpath(await mkdtemp(path.join(os.tmpdir(), "dev-loops-nogit-ledger-")));
+  try {
+    const roots = resolveLedgerCheckouts(dir);
+    assert.deepEqual(roots, [dir]);
+  } finally {
+    await rm(dir, { recursive: true, force: true });
+  }
+});
```
[the remark this answers is not in the document]
Comment (mfittko) [reply test/loop/repo-root-resolver.test.mjs]: Fixed in 62886b6d. Renamed the not-a-git-repo tests to say so accurately, and added true git-unavailable cases: resolveRepoRoot(dir, { gitCommand: "definitely-not-git-<ts>" }) falls back to dir, and resolveLedgerCheckouts(dir, { gitCommand: "<nonexistent>" }) returns [dir] via its resolveRepoRoot fallback. Both the not-a-repo and truly-unavailable (ENOENT/exec-failure) paths are now covered.

File: scripts/loop/detect-pr-gate-coordination-state.mjs [hunk 5]
```diff
@@ -262,58 +276,117 @@ async function fetchIssueBody({ repo, issue }, { env = process.env, ghCommand =
...
+      linkedIssues,
+      refinedIssues,
+      reason: `Failed to fetch body for linked issue(s) ${scopeLabel}; refinement status is unknown.`,
     };
   }
-  const { detectIssueRefinementArtifact } = await import("@dev-loops/core/loop/issue-refinement-artifact");
-  const artifact = detectIssueRefinementArtifact({ body, issueNumber: linkedIssue });
+  // Mixed branch: not allFailed, so at least one body fetched but none is
+  // refined. Report against the first successfully-fetched (non-null) issue —
+  // `evaluated[0]` may be a failed fetch (null artifact) and has no fields.
```
[the remark this answers is not in the document]
Comment (mfittko) [reply scripts/loop/detect-pr-gate-coordination-state.mjs]: Fixed in 63af256a. Reworded the comment: the failed-fetch entry still retains its `issue` field but has `artifact: null` (body fetch / artifact detection failed), rather than "has no fields". Comment wording only, no logic change.

**Comment by mfittko** [comment]
### Gate review: `pre_approval_gate`

**Reviewed head SHA:** `539e8ff9fd8f95afcae12b018892da446b1634b1`
**Verdict:** clean
**Execution mode:** fanout_fanin

**Findings summary:** 3 angles reviewed; 4 findings (see per-angle breakdown below).; Copilot review rounds exhausted (2/2); current head has zero unresolved threads and green or credibly green CI, so pre_approval_gate fallback is allowed without another Copilot re-request.

- `acceptance-criteria` → clean
  - [worth-fixing-now] detect-checkpoint-evidence.mjs:395 config load used raw cwd not resolveRepoRoot (internal inconsistency); fixed to route through resolveRepoRoot. — _accepted-for-fix_
  - [defer] inspect-run-viewer/server.mjs cwd reads missing from audit list; added to JUSTIFIED (operator-launched viewer, cwd=launch dir). — _accepted-for-fix_
- `deep` → clean
  - [defer] Ledger in a non-registered/pruned worktree would not be found; inherent to enumerate-real-worktrees design, not introduced by this PR. — _deferred_
- `yagni-dry-kiss-srp` → clean
  - [worth-fixing-now] Dead singular resolveLinkedIssueFromPr (zero non-test callers) kept alive only by its own test; deleted function + its test rows. — _accepted-for-fix_

**Next action:** await final human approval, then merge
